src/mock.js

```diff
@@ -82,6 +82,17 @@ function addMockFunctionsToSchema({ schema, mocks = {}, preserveResolvers = fals
     return customMock;
   }
 
+  function assignResolveType(type) {
+    const fieldType = getNullableType(type);
+    const namedFieldType = getNamedType(fieldType);
+
+    if (namedFieldType instanceof GraphQLUnionType ||
+        namedFieldType instanceof GraphQLInterfaceType
+    ) {
+      namedFieldType.resolveType = (data, context, info) => info.schema.getType(data.typename);
```

Comment:
Yeah, that seems like a great solution! I didn't know that `resolveType` isn't undefined by default.